configure.ac

```
@@ -95,6 +95,18 @@ AC_DEFUN([PY_CHECK_EMSCRIPTEN_PORT], [
   AS_VAR_POPDEF([py_libs])
 ])
 
+dnl PY_CHECK_CONSTANT(TYPE, CONSTANT, [INCLUDES])
+dnl Checks whether the constant exists and is actually usable (unlike
+dnl AC_CHECK_DECL that only checks if the constant exists)
+AC_DEFUN([PY_CHECK_CONSTANT], [
+  AC_MSG_CHECKING([for $2])
+  AC_COMPILE_IFELSE([AC_LANG_PROGRAM([$3], [[$1 val = $2;]])],
+    [AC_MSG_RESULT([yes])
+     AC_DEFINE([HAVE_$2], [1],
+               [Defined if $2 is usable])],
```

```suggestion
     AC_DEFINE([HAVE_$2], [1], [Defined if $2 is usable])],
```

(Sorry, but I really want to avoid vertically expanding our configure.ac as much as possible; ideally we should have all "functions" in a separate file but we don't do that...)